src/bootstrap/src/core/config/config.rs

```diff
@@ -394,6 +396,29 @@ impl SplitDebuginfo {
     }
 }
 
+/// Stack protector mode for compiling rustc itself
+#[derive(Default, Clone, PartialEq, Debug)]
+pub enum StackProtector {
+    #[default]
+    None,
+    Basic,
+    Strong,
+    All,
+}
+
+impl std::str::FromStr for StackProtector {
+    type Err = String;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "none" => Ok(StackProtector::None),
+            "basic" => Ok(StackProtector::Basic),
+            "strong" => Ok(StackProtector::Strong),
+            "all" => Ok(StackProtector::All),
```

Comment:
I'd prefer to avoid duplicating the names of the values for this unstable option here from rustc. Can we just pass this through as an opaque string?

Reply:
Modified according to comments.